packages/cli/src/rpc/file/update.ts

```diff
@@ -83,51 +78,64 @@ async function updateFile(
 
   await awaitCompletion(2, 30000);
 
-  if (status?.step !== ExplainRpc.Step.COMPLETE) {
-    warn(`[file-update] Failed to update ${file}`);
-    return {
-      exists: true,
-      succeeded: false,
-    };
-  }
+  if (status?.step !== ExplainRpc.Step.COMPLETE) throw new Error(`completion failed`);
 
-  const result = JSON.parse((status.explanation || []).join('')) as {
+  return JSON.parse((status.explanation || []).join('')) as {
     original: string;
     modified: string;
   };
+}
 
+async function updateFile(
+  navieProvider: INavieProvider,
+  file: string,
+  modified: string,
+  original: string
+): Promise<FileRpc.UpdateResponse> {
   try {
-    await applyFileUpdate(file, result.original, result.modified);
+    await applyFileUpdate(file, original, modified);
   } catch (e) {
     reportError('update file', e, file);
 
     return {
       exists: true,
       succeeded: false,
-      original: result.original,
-      modified: result.modified,
+      original,
+      modified,
     };
   }
 
   return {
     exists: true,
     succeeded: true,
-    original: result.original,
-    modified: result.modified,
+    original,
+    modified,
   };
 }
 
 export async function handler(
   navieProvider: INavieProvider,
   file: string,
-  modified: string
+  target: string,
```

Comment:
I'm finding the name `target` a bit hard to understand. It looks like this is the same as `modified`?

Reply:
It could be, or it could be a target snippet. I wanted to avoid the ambiguity of it being either an illustrative snippet with ellipses etc. or verbatim modified code.